homeassistant/config_entries.py

```diff
@@ -807,12 +806,21 @@ async def async_set_disabled_by(
         entry.disabled_by = disabled_by
         self._async_schedule_save()
 
+        dev_reg = device_registry.async_get(self.hass)
+        ent_reg = entity_registry.async_get(self.hass)
+
+        if not entry.disabled_by:
+            # The config entry is no longer disabled, enable devices and entities
+            device_registry.async_config_entry_disabled_by_changed(dev_reg, entry)
+            entity_registry.async_config_entry_disabled_by_changed(ent_reg, entry)
+
         # Unload the config entry, then fire an event
```

Comment:
We're reloading the entry below.

Reply:
If we're enabling a config entry, we should do it before or else we're going to reload the config entry afterwards.

Reply:
A loaded config entry will be reloaded if any of their devices/entities get enabled.

Reply:
Oh, I see now that it's about the comment. Let's drop it.

```suggestion
```

Reply:
Comment updated